contracts/modules/cover/Cover.sol

```diff
@@ -521,14 +538,19 @@ contract Cover is ICover, MasterAwareV2, IStakingPoolBeacon {
 
     // [todo] handle the creation of NFT 0 which is the default NFT owned by the pool manager
     return CoverUtilsLib.createStakingPool(
```

Comment:
The purpose of this library/function was to move the proxy bytecode out of the Cover contract to keep it small but we ended up moving a lot of logic in there, and the number of parameters keeps growing. I'd opt for having minimal required code in the library and keep the core logic here. This will also avoid some redundant memory copying.

Reply:
purpose more generally is to reduce bytecode size in Cover.sol if we can move logic there so it stays under 24k

so i intentionally moved logic there